staging/src/k8s.io/cli-runtime/pkg/resource/dry_run_verifier_test.go

```diff
@@ -154,3 +155,58 @@ func TestDryRunVerifierNoOpenAPI(t *testing.T) {
 		t.Fatalf("MyCRD doesn't support dry-run, yet no error found")
 	}
 }
+
+func TestHasSupportCachesOpenAPI(t *testing.T) {
+	t.Parallel()
+
+	podDryRunSupportingOpenAPI := &openapi_v2.Document{Paths: &openapi_v2.Paths{
+		Path: []*openapi_v2.NamedPathItem{{Value: &openapi_v2.PathItem{
+			Patch: &openapi_v2.Operation{
+				Parameters: []*openapi_v2.ParametersItem{{Oneof: &openapi_v2.ParametersItem_Parameter{
+					Parameter: &openapi_v2.Parameter{Oneof: &openapi_v2.Parameter_NonBodyParameter{NonBodyParameter: &openapi_v2.NonBodyParameter{Oneof: &openapi_v2.NonBodyParameter_QueryParameterSubSchema{
+						QueryParameterSubSchema: &openapi_v2.QueryParameterSubSchema{Name: "dryRun"},
+					}}}},
+				}}},
+				VendorExtension: []*openapi_v2.NamedAny{{
+					Name:  "x-kubernetes-group-version-kind",
+					Value: &openapi_v2.Any{Yaml: `{"kind":"Pod","version":"v1"}`},
+				}},
+			},
+		}}},
+	}}
+	testCases := []struct {
+		name         string
+		cachedSchema *openapi_v2.Document
+		fetchErr     error
+	}{
+		{
+			name: "No cache, schema is fetched",
+		},
+		{
+			name:         "Cached schema is used",
+			cachedSchema: podDryRunSupportingOpenAPI,
+			fetchErr:     errors.New("Thou shall use the cache!"),
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
nit: your test cases are not parallel

Reply:
The top-level test is. I haven't benchmarked it but I would strongly assume that parallelizing the subtests would actually slow things down due to the runtime overhead of starting a new goroutine ;)